### ISSUE TITLE
gunicorn service and exec

### ISSUE THREAD

**Issue by solud** [echocat/puppet-graphite#277]
CentOS 7, Puppet 3.8, gr_web_server='wsgionly'
I've had to use the following changes to get this module to work. I'm not experiencing a race condition so removing the exec cleans up the puppet output. The Exec executes on every run and the wsgi.py doesn't seem to be needed. 

https://github.com/echocat/puppet-graphite/compare/develop...solud:develop?expand=1

What does the race condition fix's script actually do and why would this need to run every time?

**Comment by dwerder** [comment]
That PR introduced the race condition exec: https://github.com/echocat/puppet-graphite/pull/237

**Comment by squarebracket** [comment]
I just submitted PR-281 which makes the race condition script only run on initial install.

The script fixes a [race condition that's present in the graphite 0.9.x branch](https://github.com/graphite-project/graphite-web/issues/403), which is the default branch used when installing via pip. That bug has been closed in the upstream graphite (v 10.0, not yet released), but no fix was backported.

The bug is caused by multiple threads trying to create a default user at the same time. The script is essentially a single-threaded call to the script snippet in graphite that causes the race condition. Without it, the web server may fail to start on first run. It will work on subsequent puppet runs because the first run, despite failing to launch properly, will create the default user/profile needed by graphite.
